python/lsst/ts/observatory/control/base_calsys.py

```diff
@@ -0,0 +1,184 @@
+            log=log,
+            intended_usage=intended_usage,
+        )
+
+        self.instrument_setup_attributes = set(instrument_setup_attributes)
+
+    async def setup_calsys(self, calib_type: str) -> str:
+        """Calibration instrument is prepared so that illumination source
+        can be turned ON. Initial instrument settings are configured
+        based on calib_type.
+
+        Parameters
+        ----------
+        calib_type : `str`
+            White Light or Monochromatic illumination system: ['White','Mono']
+
+        Raises
+        -------
+        RuntimeError:
+            If setup is unable to complete.
+
+        Notes
+        -----
+        The dome and telescope can be moving into place concurrently.
+
+        """
+        raise NotImplementedError()
+
+    async def configure_flat(
+        self,
+        filter: str,
+        wavelength: float,
+    ) -> str:
+        """Configure calibration system to be ready to take a flat
+
+        Parameters
+        ----------
+        filter : `str`
+            Rubin Filter: ['u','g','r','i','z','y']
+        wavelength : `float`
+            (units = nm)
+            If monochromatic flats, wavelength range [350, 1200] nm
+
+        Raises
+        -------
+        RuntimeError:
+            If error in configuration
+
+        Notes
+        -----
+        This will include the check the configuration of the telescope, dome,
+        and camera in addition to calsys.
+        """
+        raise NotImplementedError()
+
+    async def perform_flat(self, exptime_dict: dict) -> str:
+        """Configure calibration system to be ready to take a flat
+
+        Parameters
+        ----------
+        exptime_dict : `dict`
+            This will contain exposure times for the camera, electrometer,
+            and fiber spectrographs based on the light source, filter and
+            wavelength.
+
+        Raises
+        -------
+        RuntimeError:
+            If there is some error in the data taking of the camera,
+            electrometer or fiber spectrographs
+        """
+        raise NotImplementedError()
+
+    def check_kwargs(self, **kwargs: typing.Union[int, float, str, None]) -> None:
+        """Utility method to verify that kwargs are in
+        `self.instrument_setup_attributes`.
+
+        Parameters
+        ----------
+        **kwargs
+            Optional keyword,value pair.
+
+        Raises
+        ------
+        RuntimeError:
+            If keyword in kwargs is not in `self.instrument_setup_attributes.`
+        """
+
+        for key in kwargs:
+            if key not in self.instrument_setup_attributes:
+                raise RuntimeError(
+                    f"Invalid argument {key}."
+                    f" Must be one of {self.instrument_setup_attributes}."
+                )
+
+    @abc.abstractmethod
+    async def get_optimized_exposure_times(
+        self, calib_type: str, filter: str, wavelength: float
+    ) -> str:
```

Comment:
The documented return type is a dict but here you say it is a str. I think this is supposed to be a dict with str key and float values right? So this should be `-> dict[str, float]`.